src/views/communities/index.tsx

```diff
@@ -181,37 +181,45 @@ function CommunitiesScreen() {
         );
     };
 
+    const ITEM_HEIGHT = 147;
+
+    const getItemLayout = useCallback(
+        (data, index) => ({
+            length: ITEM_HEIGHT,
+            offset: ITEM_HEIGHT * index,
+            index,
+        }),
+        []
+    );
+
+    const renderItem = useCallback(
```

Comment:
do we really need useCallback here? And below.
If so, why?

Reply:
Performance improvements. RN thinks that the props which has inline function is a new value every-time and then it causes unnecessary re-renders. With the useCallback hook we memoize the function guaranteeing the function will not be re-rendered.

Reply:
Does this work fine as the user scroll?

Reply:
yes, it works ...